File: MoneyFlowUI/src/app/shared/dtos/DTO_Card.ts
export class DTO_Card {
    id = 0;
    userId = 0;
    name = '';
    cardType = 'Debit';
    cardTypeLabel = 'Débito';
    brand: string | null = null;
    last4Digits: string | null = null;
    maskedNumber = 'Sem final';
    creditLimit: number | null = null;
    closingDay: number | null = null;
    dueDay: number | null = null;
    isActive = true;
    createdAt?: Date | string | null = null;
}

File: MoneyFlowAPI/Application/DTOs/DTO_Card.cs
namespace MoneyFlowAPI.Application.DTOs
{
    public class DTO_Card
    {
        public int Id { get; set; }

        public int UserId { get; set; }

        public string Name { get; set; } = string.Empty;

        public string CardType { get; set; } = string.Empty;

        public string CardTypeLabel { get; set; } = string.Empty;

        public string? Brand { get; set; }

        public string? Last4Digits { get; set; }

        public string MaskedNumber { get; set; } = "Sem final";

        public decimal? CreditLimit { get; set; }

        public int? ClosingDay { get; set; }

        public int? DueDay { get; set; }

        public bool IsActive { get; set; }

        public DateTime? CreatedAt { get; set; }
    }
}
